keps/sig-auth/3257-cluster-trust-bundles/README.md

```diff
@@ -739,15 +813,36 @@ in back-to-back releases.
 
 ### Upgrade / Downgrade Strategy
 
-At present, there are no upgrade / downgrade concerns.  The ClusterTrustBundle
-feature gate controls overall availability of the feature.
+The feature is gated by these feature flags:
+- kube-apiserver
+    - `ClusterTrustBundle` controls availability of the ClusterTrustBundle API and
+      presence of the relevant rules in cluster roles for the kubelet and KCM.
+    - `ClusterTrustBundleProjection` controls availability of the `ClusterTrustBundle`
+      projected volume source in the Pod API.
+- kubelet, KCM
+    - watch API discovery for the ClusterTrustBundle API and enable the feature
+      internally once the API gets available.
+    - to disable the feature after the `ClusterTrustBundle` feature gate disables
+      the API availability, it is necessary to restart both the kubelet and KCM
+
+The feature is enabled by the kube-apiserver feature gates. Aside from enabling the API, the `ClusterTrustBundle`
+feature gate also creates the necessary rules in the `system:node` and `system:controller:root-ca-cert-publisher` cluster roles.
 
 ### Version Skew Strategy
 
-Both kubelet and kube-apiserver will need to be at 1.29 for the full featureset
-to be present.  If only kube-apiserver is at 1.29 and kubelet is lower, then the
-the pod mounting feature will be cleanly unavailable, but all other aspects of
-the feature will work.
+The ClusterTrustBundle volume projection was implemented in 1.29 and kubelet would fail to mount CTB
+volumes if it was requested via the Pod API while the feature gate is disabled on
+kubelet side. This means that pods will fail to become ready in version-skewed environments where the
+`ClusterTrustBundleProjection` kubelet feature gate is disabled, independently of the
+API version.
+
+As long as the kubelet's
+expected API version is not present, pods trying to mount a ClusterTrustBundle
+will fail to become ready as kubelet won't be able to create the mount.
+
+While the ClusterTrustBundle API is served at a different version than the KCM
+expects, the KCM won't be creating the new signer's ClusterTrustBundle object.
+Per docs this version skew situation should only ever occur during a live upgrade.
```

Comment:
Since KAS is updated first, this should be fine right?

Reply:
I'd like to think so :+1: